x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/utils/ingest/PipelineRegistry.java

```diff
@@ -85,9 +86,18 @@ protected boolean isClusterReady(ClusterChangedEvent event) {
             return false;
         }
 
+        if (getMinClusterVersion() != null && event.state().nodes().getMinNodeVersion().onOrBefore(getMinClusterVersion())) {
+            // Not all node satisfies the minimum version requirement: skipping install.
+            return false;
+        }
+
         return true;
     }
 
+    protected Version getMinClusterVersion() {
```

Comment:
Let's make it an abstract function to ensure that it is not an after thought? Can you also add a comment to describe why it's needed?

Reply:
The name is also a bit misleading, maybe `getMinSupportedNodeVersion`?